homeassistant/components/tuya/number.py

```diff
@@ -124,6 +124,18 @@
             entity_category=EntityCategory.CONFIG,
         ),
     ),
+    # Smart Water Timer
+    # No documentation available - https://www.moeshouse.com/products/smart-bluetooth-sprinkler-water-timer-with-rain-delay-filter-washer-programmable-irrigation-timer-with-automatic-and-manual-watering-for-standard-outdoor-garden-faucet-lawn-bluetooth-hub-required-for-irrigation-history
+    "sfkzq": (
+        # Number of seconds for the watering session. The device will automatically turn off after this time
+        NumberEntityDescription(
```

Comment:
Does this number entity actually count down during watering?

Reply:
yes

Reply:
That is not allowed in a number field. The number field can be a setting, but it should not be used as countdown. To have a "countdown" a sensor should be used with the timestamp device class indicating the time the countdown would finish.

Reply:
I agree with you. But the way of working of this parameter is: the user sets the initial value of the countdown and the device gradually decreases the value. When it reaches 0 the switch turns off.
This number field is used to set the initial value of the countdown

Reply:
> This number field is used to set the initial value of the countdown

What is unclear to me: Does it actually show its is counting down in the number field? As in, does the number value show the time left?

Reply:
Yes, it shows the time remaining before it turns off automatically. It is updated by the device every 15 seconds (multiples of 14 + n15). But the initial value must be sent via set command.
I attach communication log with the device

![log](https://user-images.githubusercontent.com/795322/176624001-acc55f1a-08ce-4690-8c0c-f496a4c7d2c7.png)

Reply:
Sure by that means the number entity changes every 15 seconds.

As said in a comment above, that is not allowed, we don't allow for countdowns in our entities.

Reply:
ok, I set the entity as a sensor